src/us/kbase/workspace/database/mongo/S3ClientWithPresign.java

```diff
@@ -131,17 +147,21 @@ public void presignAndPutObject(
 			// other S3 instances) the connection dies. If the stream is pretty small,
 			// you can get an error back.
 			final CloseableHttpResponse res = httpClient.execute(htp);
-			if (res.getStatusLine().getStatusCode() > 399) {
-				final byte[] buffer = new byte[1000];
-				try (final InputStream in = res.getEntity().getContent()) {
-					new DataInputStream(in).readFully(buffer);
-				} catch (EOFException e) {
-					// do nothing
+			try {
+				if (res.getStatusLine().getStatusCode() > 399) {
+					final byte[] buffer = new byte[1000];
+					try (final InputStream in = res.getEntity().getContent()) {
+						new DataInputStream(in).readFully(buffer);
```

Comment:
Noat that line 158 is covered by tests, which means that 154 must be covered by tests, since the buffer is used there